pyanaconda/modules/subscription/subscription_interface.py

```diff
@@ -19,11 +19,60 @@
 #
 from pyanaconda.modules.common.constants.services import SUBSCRIPTION
 from pyanaconda.modules.common.base import KickstartModuleInterface
-from pyanaconda.dbus.interface import dbus_interface
+from pyanaconda.modules.common.structures.subscription import SystemPurposeData
+from dasbus.server.interface import dbus_interface
+from dasbus.server.property import emits_properties_changed
+from dasbus.typing import *  # pylint: disable=wildcard-import
 
 
 @dbus_interface(SUBSCRIPTION.interface_name)
 class SubscriptionInterface(KickstartModuleInterface):
     """DBus interface for the Subscription module."""
 
+    def connect_signals(self):
+        super().connect_signals()
+        self.watch_property("SystemPurposeData", self.implementation.system_purpose_data_changed)
 
+    def GetValidRoles(self) -> List[Str]:
+        """Return all valid system purpose roles.
+
+        These are OS release specific, but could look like this:
+
+        "Red Hat Enterprise Linux Server"
+        "Red Hat Enterprise Linux Workstation"
+        "Red Hat Enterprise Linux Compute Node"
+        """
+        return self.implementation.valid_roles
+
+    def GetValidSLAs(self) -> List[Str]:
+        """Return all valid system purpose SLAs.
+
+        These are OS release specific, but could look like this:
+
+        "Premium"
+        "Standard"
+        "Self-Support"
+        """
+        return self.implementation.valid_slas
+
+    def GetValidUsageTypes(self) -> List[Str]:
+        """List all valid system purpose usage types.
+
+        These are OS release specific, but could look like this:
+
+        "Production",
+        "Development/Test",
+        "Disaster Recovery"
+        """
+        return self.implementation.valid_usage_types
+
+    @property
+    def SystemPurposeData(self) -> Structure:
+        """Return DBus structure holding current system purpose data."""
+        return SystemPurposeData.to_structure(self.implementation.system_purpose_data)
+
+    @emits_properties_changed
+    def SetSystemPurposeData(self, system_purpose_data: Structure):
+        """Set a new DBus structure holding system purpose data."""
```

Comment:
It would be great to point what DBus structure you are expecting. From the DBus point of view it could be any dictionary.

Reply:
I'll add a docstring mentioning that DBus structure corresponding to/parseable by the SystemPurposeData is expected.